helpers/remoteoauth/token.go

```diff
@@ -0,0 +1,165 @@
+package remoteoauth
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+
+	cloudquery_api "github.com/cloudquery/cloudquery-api-go"
+	"github.com/google/uuid"
+	"golang.org/x/oauth2"
+)
+
+type Token struct {
+	AccessToken string
```

Comment:
These 3 first properties are logically grouped as a `oauth2.Token` and returned as such. Maybe it makes sense for the `Token` struct to just have a `oauth2.Token` inside, instead of the 3 properties.